packages/central-server/src/apiV2/import/importSurveyResponses/importSurveyResponses.js

```diff
@@ -168,32 +168,56 @@ export async function importSurveyResponses(req, res) {
       for (let columnIndex = minSurveyResponseIndex; columnIndex <= maxColumnIndex; columnIndex++) {
         const columnHeader = getColumnHeader(sheet, columnIndex);
         const importMode = getImportMode(columnHeader);
+        const entityCode = getInfoForColumn(sheet, columnIndex, 'Entity Code');
+        const entityName = getInfoForColumn(sheet, columnIndex, 'Entity Name');
+        const entity = await models.entity.findOne({ code: entityCode });
+
+        if (entityCode && entityName) {
+          if (entity) {
+            if (entity?.name !== entityName) {
+              throw new Error(
+                `Entity code and name don't match: ${entity?.name} and ${entityName}`,
+              );
+            }
+          } else {
+            throw new Error(`Entity code does match any existing entity: ${entityCode}`);
+          }
+        }
 
         if (IMPORT_BEHAVIOURS[importMode].shouldGenerateIds) {
-          surveyResponseIds[columnIndex] = generateId();
+          surveyResponses[columnIndex] = { surveyResponseId: generateId(), entityId: entity?.id };
           isGeneratedIdByColumnIndex[columnIndex] = true;
         } else if (IMPORT_BEHAVIOURS[importMode].shouldUpdateExistingResponses) {
           const { surveyResponseId } = await getExistingResponseData(columnIndex);
 
           if (surveyResponseId) {
-            surveyResponseIds[columnIndex] = surveyResponseId;
+            surveyResponses[columnIndex] = {
+              surveyResponseId: surveyResponseId,
+              entityId: entity?.id,
+            };
           } else {
             // A matching existing response was not found, generate a new id
-            surveyResponseIds[columnIndex] = generateId();
+            surveyResponses[columnIndex] = {
+              surveyResponseId: generateId(),
+              entityId: entity?.id,
+            };
             isGeneratedIdByColumnIndex[columnIndex] = true;
           }
         } else {
-          surveyResponseIds[columnIndex] = columnHeader;
+          surveyResponses[columnIndex] = {
+            surveyResponseId: columnHeader,
+            entityId: entity?.id,
+          };
         }
       }
-      updatePersistor.setupColumnsForSheet(tabName, surveyResponseIds);
+      updatePersistor.setupColumnsForSheet(tabName, surveyResponses);
 
       for (let columnIndex = minSurveyResponseIndex; columnIndex <= maxColumnIndex; columnIndex++) {
         const columnHeader = getColumnHeader(sheet, columnIndex);
         validateColumnHeader(columnHeader, columnIndex, tabName);
 
         if (isGeneratedIdByColumnIndex[columnIndex]) {
-          const surveyResponseId = surveyResponseIds[columnIndex];
+          const surveyResponseId = surveyResponses[columnIndex]?.surveyResponseId;
```

Comment:
Will there ever be a case where `surveyResponses[columnIndex]` isn't set? If not, I think this is a nice spot to use destructuring to get` surveyResponseId`, although that is a personal preference.